src/saveSvgAsPng.js

```diff
@@ -365,15 +367,26 @@
         window.open(uri, '_temp', 'menubar=no,toolbar=no,status=no');
       }
     }
+    return;
   };
 
   out$.saveSvg = (el, name, options) => {
-    requireDomNode(el);
-    out$.svgAsDataUri(el, options || {}, uri => out$.download(name, uri));
+    return new Promise((resolve, reject) => {
+      requireDomNode(el);
+      out$.svgAsDataUri(el, options || {}, uri => {
+        out$.download(name, uri)
+        resolve();
+      });
+    });
   };
 
   out$.saveSvgAsPng = (el, name, options) => {
-    requireDomNode(el);
-    out$.svgAsPngUri(el, options || {}, uri => out$.download(name, uri));
+    return new Promise((resolve, reject) => {
+      requireDomNode(el);
+      out$.svgAsPngUri(el, options || {}, uri => {
```

Comment:
I think `svgAsPngUri` already returns a promise.

Reply:
Yes, but `download` does not, and I need to make sure that my callback fires after the entire stack is called.